### PR TITLE
Move the dependency injection classes under Google.Ads.GoogleAds.Exte…

### LINKED ISSUE #585: "Missing required parameter: refresh_token" after v21
**Describe the bug:**

Calls failing with the last release. This is a continuation of what i mentioned here;
https://github.com/googleads/google-ads-dotnet/pull/573#issuecomment-2472906991

```
System.AggregateException
  HResult=0x80131500
  Message=One or more errors occurred. (Status(StatusCode="Internal", Detail="Error starting gRPC call. TokenResponseException: Error:"invalid_request", Description:"Missing required parameter: refresh_token", Uri:""", DebugException="Google.Apis.Auth.OAuth2.Responses.TokenResponseException: Error:"invalid_request", Description:"Missing required parameter: refresh_token", Uri:"""))
  Source=Google.Ads.Gax
  StackTrace:
   at Google.Ads.Gax.Interceptors.UnaryRpcInterceptor.<>c__DisplayClass1_0`2.<Intercept>b__0(Task`1 oldTask)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Google.Api.Gax.Grpc.ApiCall.GrpcCallAdapter`2.<>c__DisplayClass4_0.<<CallAsync>g__WaitAndCallHandlers|0>d.MoveNext()
   at Google.Api.Gax.Grpc.ApiCallRetryExtensions.<>c__DisplayClass0_0`2.<<WithRetry>b__0>d.MoveNext()
   at OPG.Jobs.Ads.GoogleAdsSyncJob.<OnRun>d__3.MoveNext() in C:\Repos\dotnet-api\OPG.Jobs\Ads\GoogleAdsSyncJob.cs:line 54

  This exception was originally thrown at this call stack:
    Google.Apis.Auth.OAuth2.Responses.TokenResponse.FromHttpResponseAsync(System.Net.Http.HttpResponseMessage, Google.Apis.Util.IClock, Google.Apis.Logging.ILogger)
    Google.Apis.Auth.OAuth2.Requests.RequestExtensions.PostFormAsync(object, System.Net.Http.HttpClient, string, System.Net.Http.Headers.AuthenticationHeaderValue, Google.Apis.Util.IClock, Google.Apis.Logging.ILogger, System.Threading.CancellationToken)
    Google.Apis.Auth.OAuth2.Flows.AuthorizationCodeFlow.FetchTokenAsync(string, Google.Apis.Auth.OAuth2.Requests.TokenRequest, System.Threading.CancellationToken)
    Google.Apis.Auth.OAuth2.Flows.AuthorizationCodeFlow.FetchTokenAsync(string, Google.Apis.Auth.OAuth2.Requests.TokenRequest, System.Threading.CancellationToken)
    Google.Apis.Auth.OAuth2.Flows.AuthorizationCodeFlow.RefreshTokenAsync(string, string, System.Threading.CancellationToken)
    Google.Apis.Auth.OAuth2.UserCredential.RefreshTokenAsync(System.Threading.CancellationToken)
    Google.Apis.Auth.OAuth2.TokenRefreshManager.RefreshTokenAsync()
    Google.Apis.Auth.OAuth2.TokenRefreshManager.GetAccessTokenForRequestAsync(System.Threading.CancellationToken)
    Google.Apis.Auth.OAuth2.UserCredential.GetAccessTokenWithHeadersForRequestAsync(string, System.Threading.CancellationToken)
    Grpc.Auth.GoogleAuthInterceptors.FromCredential.AnonymousMethod__0(Grpc.Core.AuthInterceptorContext, Grpc.Core.Metadata)
    ...
    [Call Stack Truncated]

Inner Exception 1:
RpcException: Status(StatusCode="Internal", Detail="Error starting gRPC call. TokenResponseException: Error:"invalid_request", Description:"Missing required parameter: refresh_token", Uri:""", DebugException="Google.Apis.Auth.OAuth2.Responses.TokenResponseException: Error:"invalid_request", Description:"Missing required parameter: refresh_token", Uri:""")

Inner Exception 2:
TokenResponseException: Error:"invalid_request", Description:"Missing required parameter: refresh_token", Uri:""

```

As mentioned in the PR there should now be a new way to register the client via DI - the extension package is still not released. Aside from that the latest version gives me this bug. Im not sure if the new di stuff fixes this or not.

**Steps to Reproduce:**
Please provide a code snippet or steps to consistently reproduce the experienced behavior.
[DO NOT INCLUDE ANY PERSONAL OR SENSITIVE DATA IN YOUR SUBMITTED CODE]

Startup.cs
`services.AddSingleton<GoogleAdsClient>();`

GoogleAdsSyncJob.cs
``` csharp
var customerService = _client.GetService(Services.V17.CustomerService);

var listAccessibleCustomersResponse = await customerService
    .ListAccessibleCustomersAsync(new ListAccessibleCustomersRequest(), CallSettings.FromCancellationToken(cancellationToken));
```

**Expected behavior:**

**Client library version and API version:**
Client library version: [v21.1.1](https://github.com/googleads/google-ads-dotnet/blob/HEAD/CHANGELOG.md#2111) and [v21.1.0](https://github.com/googleads/google-ads-dotnet/blob/HEAD/CHANGELOG.md#2110)
Google Ads API version: 17
.NET version: net8
Operating system (Linux, Windows, ...) and version (if the bug is platform-specific): windows

**Request/Response Logs:**

**Anything else we should know about your project / environment**

### PR DESCRIPTION
**Pull request by Raibaz**
…nsions

Fixes #585.

### CODE REVIEW

**Review by jradcliff — CHANGES_REQUESTED** [review]

File: Google.Ads.GoogleAds.Extensions/src/DependencyInjection/GoogleAdsApiOptions.cs [hunk 1]
```diff
@@ -1,7 +1,13 @@
 namespace Google.Ads.GoogleAds.Extensions.DependencyInjection
 {
+    /// <summary>
+    ///
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply Google.Ads.GoogleAds.Extensions/src/DependencyInjection/GoogleAdsApiOptions.cs]: Done.

File: Google.Ads.GoogleAds.Extensions/tests/DependencyInjection/TestGoogleAdsClientServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -14,13 +19,13 @@ public class TestGoogleAdsClientServiceCollectionExtensions
     {
         private IGoogleAdsClient googleAdsClient;
 
-        [SetUp]
-        public void Setup()
+        [Test]
```
[the remark this answers is not in the document]
Comment (Raibaz) [reply Google.Ads.GoogleAds.Extensions/tests/DependencyInjection/TestGoogleAdsClientServiceCollectionExtensions.cs:22]: The core of the fix is that the DependencyInjection directory was in the wrong place (outside of the Google.Ads.GoogleAds.Extensions directory) so its classes were not being included in the Extensions package published on NuGet; most of the files are actually the same, minus minor cleanups like this, they have just been moved.

[the remark this answers is not in the document]
Comment (jradcliff) [reply Google.Ads.GoogleAds.Extensions/tests/DependencyInjection/TestGoogleAdsClientServiceCollectionExtensions.cs:22]: Ah, I missed that there were moves in here. Thanks for clarifying.

File: Google.Ads.GoogleAds.Extensions/src/DependencyInjection/GoogleAdsClientServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -6,6 +7,9 @@
 
 namespace Microsoft.Extensions.DependencyInjection
 {
+    /// <summary>
```
Comment (jradcliff) [inline Google.Ads.GoogleAds.Extensions/src/DependencyInjection/GoogleAdsClientServiceCollectionExtensions.cs:10]: nit: same cleanup here and in GoogleAdsConfig.cs

Comment (Raibaz) [reply Google.Ads.GoogleAds.Extensions/src/DependencyInjection/GoogleAdsClientServiceCollectionExtensions.cs:10]: Done.

**Pull request merged**